tools/fileinfo/detection/packers/petite/test.py

```diff
@@ -1,25 +1,99 @@
 from regression_tests import *
 
-class Test_001(Test):
+class Test_12(Test):
     settings = TestSettings(
         tool='fileinfo',
-        input='fact_rec.ex'
+        input='sample_petite12.ex_'
+    )
+
+    def test_correctly_analyzes_input_file(self):
+        assert self.fileinfo.succeeded
+        assert self.fileinfo.output.contains(r'.*Petite \(1\.2\)*')
+
+
+class Test_13(Test):
+    settings = TestSettings(
+        tool='fileinfo',
+        input='sample_petite13.ex_'
+    )
+
+    def test_correctly_analyzes_input_file(self):
+        assert self.fileinfo.succeeded
+        assert self.fileinfo.output.contains(r'.*Petite \(1\.3\)*')
+
+
+class Test_13a(Test):
+    settings = TestSettings(
+        tool='fileinfo',
+        input=['fact_rec.ex', 'sample_petite13a.ex_']
     )
 
     def test_correctly_analyzes_input_file(self):
         assert self.fileinfo.succeeded
         assert self.fileinfo.output.contains(r'.*Petite \(1\.3a\)*')
 
 
-class Test_002(Test):
-    settings=TestSettings(
+class Test_14(Test):
+    settings = TestSettings(
+        tool='fileinfo',
+        input='sample_petite14.ex_'
+    )
+
+    def test_correctly_analyzes_input_file(self):
+        assert self.fileinfo.succeeded
+        assert self.fileinfo.output.contains(r'.*Petite \(1\.4\)*')
+
+
+class Test_20(Test):
+    settings = TestSettings(
+        tool='fileinfo',
+        input='sample_petite20.ex_'
+    )
+
+    def test_correctly_analyzes_input_file(self):
+        assert self.fileinfo.succeeded
+        assert self.fileinfo.output.contains(r'.*Petite \(2\.0\)*')
+
+
+class Test_21(Test):
+    settings = TestSettings(
         tool='fileinfo',
-        input=[
-            'sample_petite_001.dat',
-        ],
-        args='--json'
+        input='sample_petite21.ex_'
     )
 
-    def test_petite_packer(self):
+    def test_correctly_analyzes_input_file(self):
+        assert self.fileinfo.succeeded
+        assert self.fileinfo.output.contains(r'.*Petite \(2\.1\)*')
+
+
+class Test_22(Test):
+    settings = TestSettings(
+        tool='fileinfo',
+        input='sample_petite22.ex_'
+    )
+
+    def test_correctly_analyzes_input_file(self):
+        assert self.fileinfo.succeeded
+        assert self.fileinfo.output.contains(r'.*Petite \(2\.2\)*')
+
+
+class Test_23(Test):
+    settings = TestSettings(
+        tool='fileinfo',
+        input='sample_petite23.ex_'
+    )
+
+    def test_correctly_analyzes_input_file(self):
+        assert self.fileinfo.succeeded
+        assert self.fileinfo.output.contains(r'.*Petite \(2\.3\)*')
+
+
+class Test_24(Test):
+    settings = TestSettings(
+        tool='fileinfo',
+        input='sample_petite24.ex_'
+    )
+
+    def test_correctly_analyzes_input_file(self):
         assert self.fileinfo.succeeded
-        self.assertEqual(self.fileinfo.output['tools'][0]['name'], 'Petite')
+        assert self.fileinfo.output.contains(r'.*Petite \(2\.4\)*')
```

Comment:
In future tests, consider adding `args='--json'` into `TestSettings` and then checking the packer and its version separately via JSON:
```python
self.assertEqual(self.fileinfo.output['tools'][0]['name'], 'Petite')
self.assertEqual(self.fileinfo.output['tools'][0]['version'], '2.4')
```
This is more accurate than just checking that the given regexp is matched in the complete output from `retdec-fileinfo`. Plus, we no longer need to use regular expressions, which is also a win :slightly_smiling_face:

Reply:
However, `self.fileinfo.output['tools'][0]['name'], 'Petite'` it's prone to tool order. If any new tool detection is added *before* Petite, this test is gonna fail.

Reply:
It's possible. This is why I have written _consider_ in the comment. However, if this is a test for a specific recognition of a packer (and we know that the sample is packed with that packer), maybe it would be nice to be notified when the tool order changes. It might indicate a bug on our side when there is another tool detected that has a higher priority.